### PR TITLE
TW-719 feat: display only first message display name when multi

### PR DESCRIPTION
**Pull request by imGok**
#719 
image

### CODE REVIEW

File: lib/pages/chat/events/message.dart [hunk 1]
```diff
@@ -206,7 +207,9 @@ class Message extends StatelessWidget {
                                       crossAxisAlignment:
                                           CrossAxisAlignment.start,
                                       children: [
-                                        ownMessage || event.room.isDirectChat
+                                        ownMessage ||
+                                                event.room.isDirectChat ||
+                                                !showDisplayName
```
Comment (hoangdat) [inline lib/pages/chat/events/message.dart:212]: Deserve a method

Comment (drminh2807) [reply lib/pages/chat/events/message.dart:212]: .

Comment (imGok) [reply lib/pages/chat/events/message.dart:212]: there is already one I forgot to use it mb

File: lib/pages/chat/events/message.dart [hunk 2]
```diff
@@ -579,6 +576,9 @@ class Message extends StatelessWidget {
     );
   }
 
+  bool displayUsername(bool ownMessage) =>
+      ownMessage || event.room.isDirectChat || !isSameSender(nextEvent, event);
```
Comment (hoangdat) [inline lib/pages/chat/events/message.dart:580]: no need to display name in `direct chat`

File: lib/pages/chat/events/message.dart [hunk 3]
```diff
@@ -683,26 +683,26 @@ class Message extends StatelessWidget {
...
+    }.contains(comparedEvent.type);
 
     // Ignoring events that are not messages, stickers, encrypted or redaction.
-    if (!isPreviousEventMessage) {
+    if (!isPreviousOrNextEventMessage) {
       return true;
     }
 
-    return previousEvent.senderId != currentEvent.senderId;
+    return currentEvent.senderId != comparedEvent.senderId;
```
Comment (hoangdat) [inline lib/pages/chat/events/message.dart:705]: why we need to change it?

File: lib/pages/chat/events/message.dart (hunk 2 above)
Comment (imGok) [reply lib/pages/chat/events/message.dart:580]: methods name was not good, it should be "hideDisplayName"

File: lib/pages/chat/events/message.dart (hunk 3 above)
Comment (imGok) [reply lib/pages/chat/events/message.dart:705]: I prefer to refactor this one so we can use it with previousEvent and nextEvent, didn't want to do 2 sames methods with exact same code

File: lib/pages/chat/events/message.dart [hunk 4]
```diff
@@ -680,26 +683,26 @@ class Message extends StatelessWidget {
     );
   }
 
-  // Check if the sender of the current event is the same as the previous event.
-  bool isSameSender(Event? previousEvent, Event currentEvent) {
-    // If the previous event is null, it is assumed that the message is the newest.
-    if (previousEvent == null) {
+  // Check if the sender of the current event is the same as the compared event.
+  bool isSameSender(Event? comparedEvent, Event currentEvent) {
```
Comment (sherlockvn) [inline lib/pages/chat/events/message.dart:687]: i think `previousEvent` is better name, appropriate with currentEvent name

**Review by drminh2807 — APPROVED** [review]

File: lib/pages/chat/events/message.dart (hunk 4 above)
Comment (imGok) [reply lib/pages/chat/events/message.dart:687]: it shouldn't be specific to previous or next because we use this method for both previous and next

**Review by hoangdat — APPROVED** [review]

**Pull request merged**